mon-entreprise/index.html

```diff
@@ -212,36 +215,70 @@
 			// Hack to force styled components to render styles during prerender
 			if (window.__PRERENDER_INJECTED) {
 				window.onload = () => {
-					var el = document.createElement('style');
+					var el = document.createElement('style')
 					document.head.appendChild(el)
 					var styles = document.querySelectorAll('style[data-styled]')
 					for (style of styles.values()) {
-							for (rule of style.sheet.rules) {
-								el.appendChild(document.createTextNode(rule.cssText))
-							}
+						for (rule of style.sheet.rules) {
+							el.appendChild(document.createTextNode(rule.cssText))
+						}
 					}
+				}
 			}
-			};
 		</script>
 		<!-- APP  -->
 		<div id="js"></div>
 
+		<script>
+			// Set the main colors from the provided customization in the URL We do it
+			// before loading the whole JS bundle to avoid a UI flash. cf. #1786
+			try {
+				let iframeColor = new URLSearchParams(
+					document.location.search.substring(1)
+				).get('couleur')
+				if (iframeColor) {
+					;[
+						document.documentElement,
+						...document.querySelectorAll('.js-color-element'),
+					].forEach((element) => {
+						element.style.setProperty('--color', iframeColor)
+						element.style.setProperty('--lightColor', iframeColor)
+						element.style.setProperty('--darkColor', iframeColor)
```

Comment:
Pour simplifier on utilise la même couleur pour les variations “dark” et “light” du thème. Ça me semble un compromis acceptable pour ne pas trop complexifier le code (par exemple ramener le calcul des variations dans ce script), et de toute façon c'est juste pour une seconde. On évite le flash bleu ce qui est le principal.

Comme noté dans l'issue initiale le SSR peut résoudre ce problème d'une manière plus systématique #1762

Reply:
Yes, j'avais vu, ça me semble un bon compromis en effet :)